clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/view/provider/KubernetesManifestProvider.java

```diff
@@ -69,26 +77,45 @@ public KubernetesV2Manifest getManifest(
       return null;
     }
 
+    Future<List<KubernetesManifest>> events =
+        includeEvents
+            ? executorService.submit(() -> credentials.eventsFor(coords))
+            : Futures.immediateFuture(ImmutableList.of());
+
+    Future<List<ContainerMetric>> metrics =
+        includeEvents
+                && coords.getKind().equals(KubernetesKind.POD)
+                && credentials.isMetricsEnabled()
+            ? executorService.submit(() -> getPodMetrics(credentials, coords))
+            : Futures.immediateFuture(ImmutableList.of());
+
     KubernetesManifest manifest = credentials.get(coords);
```

Comment:
Is there any reason to not also request the manifest itself on a separate thread?

Reply:
I hesitated between the two; it seems like the advice on the internet is that often it's more clear to just send all your tasks to the executor so there's no asymmetry around one of them happening on the main thread and the others happening in the executor.

But in this case since we do have something like a "parent task" (getting the manifest) and ancillary tasks (getting the metrics/events) it seemed reasonable to get the manifest on the main thread.  (This also completely avoids the need to use the executor in the case where we don't want metrics/events.).

Anyway, that was my thinking and why I leaned towards fetching the manifest on the main thread.